foxpuppet/browser/window.py

```diff
@@ -0,0 +1,96 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from .tabbar import TabBar
+from .navbar import Navbar
+from foxpuppet.browser.windows import Windows
+
+NAV_BAR = 'nav-bar'
+TAB_BROWSER = 'tabbrowser-tabs'
+TITLE_BAR_CLOSE = 'titlebar-close'
+TITLE_BAR_MIN = 'titlebar-min'
+TITLE_BAR_MAX = 'titlebar-max'
+
+
+class Window(object):
+
+    def __init__(self, selenium, *args, **kwargs):
+        self.selenium = selenium
+        self.windows = Windows(selenium)
+        self._navbar = None
+        self._tabbar = None
+
+    @property
+    def navbar(self):
+        self.selenium.switch_to.window(self.windows.current)
+
+        if not self._navbar:
+            navbar = self.selenium.find_element_by_id(NAV_BAR)
+            self._navbar = Navbar(self.selenium, navbar)
```

Comment:
Let's remove the caching and revisit if we feel there are worthwhile performance gains.